### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by chfast**
Move the build instructions from the http://ethdocs.org. Also update info about license and do some small cleanups.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,6 +29,94 @@ Please do come and chat to us on the [cpp-ethereum](https://gitter.im/ethereum/c
...
+```
+
+The `--recursive` option is important. It orders git to clone additional 
+submodules which are required to build the project.
+If you missed it you can correct your mistake with command 
+`git submodule update --init`.
+
+### Install CMake
+
+CMake us used to control the build configuration of the project. Quite recent
```
Comment (gumb0) [inline README.md]: is used

File: README.md [hunk 2]
```diff
@@ -25,6 +29,94 @@ Please do come and chat to us on the [cpp-ethereum](https://gitter.im/ethereum/c
...
+
+CMake us used to control the build configuration of the project. Quite recent 
+version of CMake is required 
+(at the time of writing [3.4.3 is the minimum](CMakeLists.txt#L25)).
+We recommend installing CMake by downloading and unpacking the binary 
+distribution  of the latest version available on the 
+[**CMake download page**](https://cmake.org/download/).
+
+The CMake package available in your operating system can also be installed
+and used if meets the minimum version required.
```
Comment (gumb0) [inline README.md]: "if it meets the minimum version requirement" ?

File: README.md [hunk 3]
```diff
@@ -25,6 +29,94 @@ Please do come and chat to us on the [cpp-ethereum](https://gitter.im/ethereum/c
...
+`build` directory with the configuration.
+
+```shell
+cmake -H. -Bbuild
+cmake --build build
+```
+
+On Windows Visual Studio 2015 is required. You should generate Visual Studio 
+solution file (.sln) for 64-bit architecture by adding 
+`-G "Visual Studio 14 2015 Win64` argument to the CMake configure command.
```
Comment (gumb0) [inline README.md]: The full command would still be clearer

File: README.md [hunk 4]
```diff
@@ -25,6 +29,94 @@ Please do come and chat to us on the [cpp-ethereum](https://gitter.im/ethereum/c
...
+and used if meets the minimum version required.
+
+> **Alternative method**
+>
+> The repository contains the
+[scripts/install_cmake.sh](scripts/install_cmake.sh) script that downloads 
+> a fixed version of CMake and unpacks it to the given directory prefix. 
+> Example usage: `scripts/install_cmake.sh --prefix /usr/local`.
+
+### Install dependencies (Linux, MacOS)
```
Comment (gumb0) [inline README.md]: MacOS is now officially known as "macOS" 😃

**Comment by chfast** [comment]
How it looks now?

**Comment by chfast** [comment]
This also bumps CMake to 3.4, and #3653 to 3.5. Please raise you objections now.

**Comment by gumb0** [comment]
"Contributing" section has a dead link to Contributing.md and an outdated link to "good first task" label

**Pull request merged**